File: lib/python/customer_wireless_speeds.py
import functions

customer_id = 917336
#get a token for our session. Using "Session" from the requests library, the token is stored on the session "object"
functions.get_token()
request = {
    "operationName": "customer",
    "query":"query customer($customer_id:Int!){\n customer(customer_id:$customer_id){\n package_instances {\n package_id\n package_name\n service_instances{\n service_details {\n ... on ServiceTypeWifi{\n service_number\n username\n up_speed\n down_speed\n up_speed_unit\n down_speed_unit\n}\n}\n}\n}\n}\n}",
    "variables": {
        "customer_id": customer_id
    }
};
#send request to API
response = functions.get_ubo_api(request)
#extract data from response
data = response["data"]["customer"]
for package_instance in data["package_instances"]:
    for service_instance in package_instance["service_instances"]:
        wifiService = service_instance["service_details"]
        if(wifiService!={} and wifiService != None):
            print("Service Username: %s" % wifiService["username"])
            print("Down speed: %s %s" % (wifiService["down_speed"],wifiService["down_speed_unit"]))
            print("Up speed: %s %s" % (wifiService["up_speed"],wifiService["up_speed_unit"]))

File: lib/golang/customer_wireless_speeds.go
package main

import (
  "fmt"
)

func main(){
  token := getToken();

  operationName := "customer"
  query := `query customer($customer_id:Int!){\n customer(customer_id:$customer_id){\n package_instances {\n package_id\n package_name\n service_instances{\n service_details {\n ... on ServiceTypeWifi{\n service_number\n username\n up_speed\n down_speed\n up_speed_unit\n down_speed_unit\n}\n}\n}\n}\n}\n}`
  variables := `{"customer_id": 1432780}`

  req := `{"operationName": "`+operationName+`","query":"`+query+`","variables": `+variables+` }`
  data := UBOApiData{
    body: req,
    method: "POST",
    token: token,
  };

  response, err := sendApiRequest(data)
  if (err != nil) {
    fmt.Println(err)
    return
  }
  customer := response["customer"].(map[string]interface{})
  //output response
  packageInstances := customer["package_instances"]
  for _ , packageInstance := range packageInstances.([]interface {}) {
    packageInstance := packageInstance.(map[string]interface{})
    serviceInstances := packageInstance["service_instances"]
    for _, serviceInstance := range serviceInstances.([]interface {}){
      serviceInstance := serviceInstance.(map[string]interface{})
      serviceDetails := serviceInstance["service_details"]
      if serviceDetails != nil {
        serviceDetails := serviceDetails.(map[string]interface{})
        username := serviceDetails["username"].(string)
        downSpeed := fmt.Sprintf("%g",serviceDetails["down_speed"].(float64)) //convert to string immediately
        downSpeedUnit := serviceDetails["down_speed_unit"].(string)
        upSpeed := fmt.Sprintf("%g",serviceDetails["up_speed"].(float64))
        upSpeedUnit := serviceDetails["up_speed_unit"].(string)

        fmt.Println("Service Username: "+username)
        fmt.Println("Down Speed: "+downSpeed+" "+downSpeedUnit)
        fmt.Println("Up Speed: "+upSpeed+" "+upSpeedUnit)
      }
    }
  }
}
